runner/types.go

```diff
@@ -0,0 +1,95 @@
+	Location         string `json:"location,omitempty" csv:"location" dsl:"location"`
+	Title            string `json:"title,omitempty" csv:"title"`
+	str              string
+	Scheme           string              `json:"scheme,omitempty" csv:"scheme" dsl:"scheme"`
+	Error            string              `json:"error,omitempty" csv:"error"`
+	WebServer        string              `json:"webserver,omitempty" csv:"webserver" dsl:"webserver"`
+	ResponseBody     string              `json:"body,omitempty" csv:"body" dsl:"body"`
+	ContentType      string              `json:"content_type,omitempty" csv:"content_type" dsl:"content_type"`
+	Method           string              `json:"method,omitempty" csv:"method" dsl:"method"`
+	Host             string              `json:"host,omitempty" csv:"host" dsl:"host"`
+	Path             string              `json:"path,omitempty" csv:"path" dsl:"path"`
+	FavIconMMH3      string              `json:"favicon,omitempty" csv:"favicon"`
+	FinalURL         string              `json:"final_url,omitempty" csv:"final_url"`
+	ResponseHeader   map[string]string   `json:"header,omitempty" csv:"header" dsl:"header"`
+	RawHeader        string              `json:"raw_header,omitempty" csv:"raw_header"`
+	Request          string              `json:"request,omitempty" csv:"request" dsl:"request"`
+	ResponseTime     string              `json:"time,omitempty" csv:"time" dsl:"time"`
+	Jarm             string              `json:"jarm,omitempty" csv:"jarm"`
+	ChainStatusCodes []int               `json:"chain_status_codes,omitempty" csv:"chain_status_codes"`
+	A                []string            `json:"a,omitempty" csv:"a" dsl:"a"`
+	CNAMEs           []string            `json:"cname,omitempty" csv:"cname"`
+	Technologies     []string            `json:"tech,omitempty" csv:"tech"`
+	Extracts         map[string][]string `json:"extracts,omitempty" csv:"extracts"`
+	Chain            []httpx.ChainItem   `json:"chain,omitempty" csv:"chain"`
+	Words            int                 `json:"words" csv:"words" dsl:"words"`
+	Lines            int                 `json:"lines" csv:"lines" dsl:"lines"`
+	StatusCode       int                 `json:"status_code,omitempty" csv:"status_code" dsl:"status_code"`
+	ContentLength    int                 `json:"content_length,omitempty" csv:"content_length" dsl:"content_length"`
+	Failed           bool                `json:"failed" csv:"failed" dsl:"failed"`
+	VHost            bool                `json:"vhost,omitempty" csv:"vhost"`
+	WebSocket        bool                `json:"websocket,omitempty" csv:"websocket"`
+	CDN              bool                `json:"cdn,omitempty" csv:"cdn"`
+	HTTP2            bool                `json:"http2,omitempty" csv:"http2"`
+	Pipeline         bool                `json:"pipeline,omitempty" csv:"pipeline"`
+}
+
+// function to get dsl variables from result struct
+func dslVariables() []string {
+	dslVarList := []string{}
```

Comment:
it'd be better to use `fatih/structs` as it automatically parses embedded/nested fields without having to go through parsing them manually as we are doing now

Reply:
I was hesitant about using https://github.com/fatih/structs because it looks like it has been abandoned and there hasn't been any updates in last 4 years.